conans/test/integration/symlinks_test.py

```diff
@@ -163,3 +163,62 @@ def upload_test(self):
         client.save({"conanfile.txt": test_conanfile}, clean_first=True)
         client.run("install conanfile.txt")
         self._check(client, ref, build=False)
+
+
+@unittest.skipUnless(platform.system() != "Windows", "Requires Symlinks")
+class ExportSymLinksTest(unittest.TestCase):
+
+    def _initialize_client(self, conanfile):
+        self.client = TestClient()
+        self.client.save({"conanfile.py": conanfile,
+                          "src/main.cpp": "cpp fake content",
+                          "CMakeLists.txt": "cmake fake content",
+                          "another_directory/not_to_copy.txt": ""})
+        self.other_dir = os.path.join(self.client.current_folder, "another_other_directory")
+        mkdir(self.other_dir)
+
+    def export_pattern_test(self):
+        conanfile = """
+from conans import ConanFile, CMake
+
+class ConanSymlink(ConanFile):
+    name = "ConanSymlink"
+    version = "3.0.0"
+    exports_sources = ["src/*", "CMakeLists.txt"]
+"""
+        self._initialize_client(conanfile)
+        os.symlink("another_directory", os.path.join(self.other_dir, "another_directory"))
+        self.client.run("export . danimtb/testing")
+        ref = ConanFileReference("ConanSymlink", "3.0.0", "danimtb", "testing")
+        export_sources = self.client.paths.export_sources(ref)
+        cache_other_dir = os.path.join(export_sources, "another_other_directory")
+        cache_src = os.path.join(export_sources, "src")
+        cache_main = os.path.join(cache_src, "main.cpp")
+        cache_cmake = os.path.join(export_sources, "CMakeLists.txt")
+        self.assertFalse(os.path.exists(cache_other_dir))
```

Comment:
According to the documentation [Python os.exists](https://docs.python.org/3/library/os.path.html#os.path.exists) : Returns False for broken symbolic links.

If my previous [comment](https://github.com/conan-io/conan/pull/3396/commits/c514a1d988d6d5f112161fd632c1c62eab7af321#r212973066) is correct, then the symbolic link is invalid, hence the assert is ok.

Maybe using `os.path.lexists()` could be better ?